### DISCUSSION TITLE
Best practices for upgrading frameworks?

### DISCUSSION THREAD

**Discussion by modir** [composer/composer#11267] [Q&A / Support] [answered]
I often come to projects where they use an outdated framework. At the moment it is often a lot of manual work because there is no command to upgrade all the a newest version as discussed in this [feature request](https://github.com/composer/composer/discussions/9588).

How do you handle such situations? Do you have some rules you always follow?

**Comment by Seldaek** [comment, marked as answer]
There is `composer bump` which can help with that, but it won't allow major upgrades. For minor versions though you could do `composer update framework/* && composer bump framework/*`.

For major versions I usually just edit it manually in composer.json, selecting all packages I want to edit with multi-cursor edit then replacing all constraints by `^2` or whatever the next major is.
